### PR TITLE
Fix #10320: Add Athena Usage & Lineage

### LINKED ISSUE #10320: Athena Linage and Usage Ingestion
**Is your feature request related to a problem? Please describe.**
We have our data lake and DWH build on top of Athena , we want to utilize OpenMetadata for which we need to track linage and queries on the table

**Describe the solution you'd like**
Similar to one we have for Redshift

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes:

Fix #10320: Add Athena Usage & Lineage

Athena does not store the query history log in a specific table but it can be fetched via calling the [list_query_executions](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/athena/client/list_query_executions.html) API which returns only the id of executed queries and [batch_get_query_execution](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/athena/client/batch_get_query_execution.html) API is used to fetch the details of those queries.

This approach is somewhat similar to what we do for databricks but the APIs does not provide the functionality to filter the queries based of datetime or query text hence the we will be dependent on the [resultLimit](https://github.com/open-metadata/OpenMetadata/blob/752163ac714e239133574a00eb6ec89dcdf28930/openmetadata-spec/src/main/resources/json/schema/metadataIngestion/databaseServiceQueryUsagePipeline.json#L30) to limit the result set

also cleaned some code in databricks usage & lineage as well

#
### Type of change:

- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

- [x] The issue properly describes why the new feature is needed, what's the goal, and how we are building it. Any discussion
    or decision-making process is reflected in the issue.
- [x] TODO: I have updated the documentation. 
- [ ] I have added tests around the new logic.

### CODE REVIEW

**Review by harshach — APPROVED** [review]

File: ingestion/src/metadata/ingestion/source/database/athena/usage.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    def yield_table_queries(self) -> Optional[Iterable[TableQuery]]:
+        """
+        Method to yield TableQueries
+        """
+        for query_list in self.get_queries() or []:
+            queries = [
+                TableQuery(
+                    query=query.Query,
+                    startTime=datetime.strftime(
+                        query.Status.SubmissionDateTime, "%Y-%m-%d %H:%M:%S.%f"
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/athena/usage.py:43]: nit - have a constant for the time format

**Review by nahuelverdugo — APPROVED** [review]
LGTM 🚀

**Review by pmbrull — APPROVED** [review]
thanks, just a quick nit if you have the time

File: ingestion/src/metadata/utils/constants.py [hunk 2]
```diff
@@ -28,3 +28,9 @@
     "useSSL": "use_ssl",
     "verifyCerts": "verify_certs",
 }
+
+QUERY_WITH_OM_VERSION = '/* {"app": "OpenMetadata"'
+
+QUERY_WITH_DBT = '/* {"app": "dbt"'
+
+DATETIME_STR_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
```
Comment (nahuelverdugo) [inline ingestion/src/metadata/utils/constants.py:36]: Should we move this:
```python
datetime.strftime(
                        query.Status.SubmissionDateTime, DATETIME_STR_FORMAT
                    )
```
into `time_utils.py`?

**Review by nahuelverdugo — APPROVED** [review]

**Pull request merged**